### PR TITLE
disable fault domain

### PR DESCRIPTION
**Pull request by mnaboka**
## High Level Description

set `fault_domain_enabled` to `false` in all cases.

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
current tests coverage is sufficient for this change
  - [X] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [X] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

File: gen/build_deploy/bash.py [hunk 1]
```diff
@@ -27,19 +27,23 @@
...
     try:
         with open(fault_domain_detect_filename):
-            pass
+            return 'true'
     except FileNotFoundError:
-        return 'false'
-    return 'true'
+        pass
+
+    return 'false'
```
Comment (branden) [inline gen/build_deploy/bash.py]: It looks like this stanza is doing the same thing as before: in both versions if the file exists it returns `'true'`, if the file doesn't exist it returns `'false'`.

And actually I think we can get rid of this stanza entirely (except for the `return 'false'` at the end). Now we're conditionally setting `fault_domain_detect_contents` based on the presence of the file, and then setting `fault_domain_enabled` based on that, so this is duplicating the file existence check that `calculate_fault_domain_detect_contents()` already does.

**Comment by darkonie** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
The enterprise bump is just for testing so I'll give this a shipit even though that doesn't have one.

**Comment by branden** [comment]
@mesosphere-mergebot label shipit

**Comment by mellenburg** [comment]
@branden @darkonie This PR is the only one on the current train and it seems that train is blowing up on upgrades. All we see is that the test SSHs to the master to run the node upgrade script, it starts, then exits 1.

One such tests has been kicked off for this PR: https://teamcity.mesosphere.io/viewLog.html?buildId=807146&buildTypeId=DcOs_Open_Test_UpgradeTest_FromStable

If that passes, then no issue.

**Comment by mellenburg** [comment]
@darkonie @amitaekbote @branden Uh-oh. Looks like this PR is the culprit indeed. This PR will be bumped back to WIP until we can get that to pass.

**Comment by darkonie** [comment]
opened an issue for core team https://jira.mesosphere.com/browse/CORE-1314

**Comment by mnaboka** [comment]
@mesosphere-mergebot label Ready For Review

File: gen/tests/test_validation.py [hunk 2]
```diff
@@ -733,3 +733,14 @@ def test_validate_custom_checks():
             'node check names: node-check-1, node-check-2.'
         ),
     )
+
+
+def test_fault_domain_disabled():
+    arguments = make_arguments(new_arguments={
+        'fault_domain_detect_filename': 'gen/fault-domain-detect/aws.sh'
```
Comment (branden) [inline gen/tests/test_validation.py]: This is a relative path, which means it depends on the test environment, which may make the test brittle. Let's use `pkg_resources.resource_filename()` to make sure we can always find it. Example use here: https://github.com/dcos/dcos/blob/0de96834fdebf56e8c8ae1ca77f4ef62b4885a5b/gen/tests/utils.py#L21

**Comment by mnaboka** [comment]
@branden updated, please review

**Comment by branden** [comment]
This looks good, enterprise counterpart at https://github.com/mesosphere/dcos-enterprise/pull/1534 is passing CI, so :ship:

**Comment by branden** [comment]
@mesosphere-mergebot label shipit

**Pull request merged**